x-pack/plugin/monitoring/src/main/java/org/elasticsearch/xpack/monitoring/exporter/http/HttpExporter.java

```diff
@@ -424,6 +425,14 @@ public HttpExporter(final Config config, final SSLService sslService, final Thre
 
         // mark resources as dirty after any node failure or license change
         listener.setResource(resource);
+
+        //for a mixed cluster upgrade, ensure that if master changes and this is the master, allow the resources to re-publish
+        onLocalMasterListener = clusterChangedEvent -> {
+            if (clusterChangedEvent.nodesDelta().masterNodeChanged() && clusterChangedEvent.localNodeMaster()) {
+                resource.markDirty();
+            }
+        };
```

Comment:
The way I read this, the resource will be marked as dirty any time the node becomes master, not just when it has become master after a mixed cluster upgrade. Am I reading that wrong or is that acceptable?

Reply:
> the resource will be marked as dirty any time the node becomes master

That is correct. However, I am not concerned about the overhead since marking it dirty it will only check to see if the resource exists (and the license level is OK). It will not re-put the watch... even it if it did, it would be no-op. This all happens on the monitoring thread too, so no worry about extra work (except setting a boolean) on the cluster state applier thread.

Reply:
Thanks! LGTM.